proposals/lip-0055.md

```diff
@@ -105,16 +115,42 @@ The full processing of a block is organized as follows.
 14. **Result verification**: Block header properties, which require accessing the state store *after* all state transitions implied by the block have been executed, are verified.
 15. **Block storage**: The block is persisted into the database.
 16. **Peers notification**: Other peers in the P2P network are notified of the new block.
 
+
+### Block Generation Stages
+
+<img src="lip-0055/generation_stages.png" width="70%">
+
+*Figure 2: A schematic depiction of the various stages of a block generation. The steps performed in the generator domain are indicated by blue boxes, while the steps performed in the state-machine domain are indicated by red boxes. Within the state-machine domain, the transactions processing is indicated by purple boxes. The network domain is indicated by green boxes.*
+
+In this section, we describe the various stages of a block generation.
+
+Similarly to the block processing, the block generation is split between the network, consensus, and state-machine domains.
+The full generation of a block is organized as follows.
+
+1. **Header creation**: Block header properties that require access to the state store *before* any state transitions implied by the block are executed are inserted in this stage.
+2. **Block initialization**: Block assets properties that require access to the state store *before* any state transitions implied by the block are executed are inserted in this stage.
+3. **Before transactions execution**: Each module can define protocol logic that is executed *before* the transactions contained in the block are processed.
+4. **Transaction verification**: Each module can define protocol logic that verifies a transaction, possibly by accessing the state store. If an error occurs, the transaction is invalid and it is not included in the block.
+5. **Command verification**: The command corresponding to the `moduleID`-`commandID` combination is verified. If an error occurs, the transaction is invalid and it is not included in the block.
+6. **Before command execution**: Each module can define protocol logic that is processed *before* the command has been executed. If an error occurs, the transaction is invalid and it is not included in the block.
+7. **Command execution**: The command corresponding to the `moduleID`-`commandID` combination is executed. If an error occurs, the transaction is failed and the processing continues.
+8. **After command execution**: Each module can define protocol logic that is processed *after* the command has been executed. If an error occurs, the transaction is invalid and it is not included in the block.
+9. **After transactions execution**: Each module can define protocol logic that is executed *after* all the transactions contained in the block have been processed.
+9. **Block sealing**: Block assets properties that require access to the state store *after* any state transitions implied by the block are executed are inserted in this stage.
```

Comment:
Similar as above, should we maybe explicitly state which module hooks are executed here?

Reply:
edited